activemq-broker/src/main/java/org/apache/activemq/plugin/SubQueueSelectorCacheBroker.java

```diff
@@ -58,39 +49,27 @@
  * https://issues.apache.org/activemq/browse/AMQ-3004
  * http://mail-archives.apache.org/mod_mbox/activemq-users/201011.mbox/%3C8A013711-2613-450A-A487-379E784AF1D6@homeaway.co.uk%3E
  */
-public class SubQueueSelectorCacheBroker extends BrokerFilter implements Runnable {
+public class SubQueueSelectorCacheBroker extends BrokerFilter {
     private static final Logger LOG = LoggerFactory.getLogger(SubQueueSelectorCacheBroker.class);
     public static final String MATCH_EVERYTHING = "TRUE";
 
     /**
      * The subscription's selector cache. We cache compiled expressions keyed
      * by the target destination.
      */
-    private ConcurrentMap<String, Set<String>> subSelectorCache = new ConcurrentHashMap<String, Set<String>>();
+    private final SubSelectorCache subSelectorCache;
 
-    private final File persistFile;
     private boolean singleSelectorPerDestination = false;
     private boolean ignoreWildcardSelectors = false;
     private ObjectName objectName;
 
-    private boolean running = true;
-    private final Thread persistThread;
-    private long persistInterval = MAX_PERSIST_INTERVAL;
-    public static final long MAX_PERSIST_INTERVAL = 600000;
-    private static final String SELECTOR_CACHE_PERSIST_THREAD_NAME = "SelectorCachePersistThread";
-
     /**
      * Constructor
      */
-    public SubQueueSelectorCacheBroker(Broker next, final File persistFile) {
+    public SubQueueSelectorCacheBroker(Broker next, final SubSelectorCache subSelectorCache) {
```

Comment:
Instead of changing the default behavior, I would rather create a totally new plugin, extending `SubQueueSelectorCacheBroker`. I think the configuration would be easier and less confusing for the users.

Reply:
+1 JB

Reply:
I can do that if you'd like, but I lean towards this approach for the usual composition-over-inheritance reasons. 

I'm interested in better understanding how this breaks previous users or changes the default behavior? The behavior is the same I believe, just factored out to a separate class.

If you're concerned about changing the constructor signature, I could add the original signature back and keep the new one.

Reply:
@jbonofre @mattrpav Thoughts appreciated! Again, happy to move forward, just wanted to take the opportunity to understand a little bit more what you meant. Thanks!

Reply:
Yes, the default behavior is the same as you "provide" the plugin already (using file persist). My point is more in term of configuration for users. But I guess it's acceptable.